packages/services/api/src/modules/audit-logs/providers/audit-logs-manager.ts

```diff
@@ -0,0 +1,152 @@
+import { Injectable, Scope } from 'graphql-modules';
+import { z } from 'zod';
+import * as Sentry from '@sentry/node';
+import { QueryAuditLogsArgs } from '../../../__generated__/types.next';
+import { User } from '../../../shared/entities';
+import { ClickHouse, sql } from '../../operations/providers/clickhouse-client';
+import { SqlValue } from '../../operations/providers/sql';
+import { Logger } from '../../shared/providers/logger';
+import { formatToClickhouseDateTime } from '../helpers';
+import { AuditLogEvent, auditLogSchema } from './audit-logs-types';
+
+export const AUDIT_LOG_CLICKHOUSE_OBJECT = z.object({
+  id: z.string(),
+  event_time: z.string(),
+  user_id: z.string(),
+  user_email: z.string(),
+  organization_id: z.string(),
+  event_action: z.string(),
+  metadata: z.string().transform(x => JSON.parse(x)),
+});
+
+export type AuditLogModel = z.infer<typeof AUDIT_LOG_CLICKHOUSE_OBJECT>;
+
+const AUDIT_LOG_CLICKHOUSE_ARRAY = z.array(AUDIT_LOG_CLICKHOUSE_OBJECT);
+
+type AuditLogRecordEvent = {
+  userId: string;
+  userEmail: string;
+  organizationId: string;
+  user: (User & { isAdmin: boolean }) | null;
+};
+
+@Injectable({
+  scope: Scope.Operation,
+  global: true,
+})
+export class AuditLogManager {
+  private logger: Logger;
+
+  constructor(
+    logger: Logger,
+    private clickHouse: ClickHouse,
+  ) {
+    this.logger = logger.child({ source: 'AuditLogManager' });
+  }
+
+  createLogAuditEvent(event: AuditLogEvent, record: AuditLogRecordEvent): void {
+    void this.internalCreateLogAuditEvent(event, record);
+  }
+
+  private async internalCreateLogAuditEvent(
+    event: AuditLogEvent,
+    record: AuditLogRecordEvent,
+  ): Promise<void> {
+    try {
+      const { eventType } = event;
+      const { organizationId, userEmail, userId } = record;
+      this.logger.debug('Creating a log audit event (event=%o)', event);
+
+      const parsedEvent = auditLogSchema.parse(event);
+      const metadata = {
+        user: record.user,
+        ...parsedEvent,
+      };
+
+      const eventMetadata = JSON.stringify(metadata);
+      const eventTime = new Date();
+
+      const values = [eventTime, userId, userEmail, organizationId, eventType, eventMetadata];
+
+      await this.clickHouse.insert({
+        query: sql`
+        INSERT INTO audit_log
+        (event_time, user_id, user_email, organization_id, event_action, metadata)
+        FORMAT CSV`,
+        data: [values],
+        timeout: 5000,
+        queryId: 'create-audit-log',
+      });
+    } catch (error) {
+      this.logger.error('Failed to create audit log event', error);
+      Sentry.captureException(error, {
+        extra: {
+          event,
+        },
+      });
+    }
+  }
+
+  async getPaginatedAuditLogs(
+    props: QueryAuditLogsArgs,
```

Comment:
Please don't use GraphQL types here, please just define the correct expected type inline, then we also don't need the eslint disables and code assertions.